parquet-hadoop/src/main/java/parquet/hadoop/InternalParquetRecordReader.java

```diff
@@ -202,12 +205,29 @@ public boolean nextKeyValue() throws IOException, InterruptedException {
 
     while (!recordFound) {
       // no more records left
-      if (current >= total) { return false; }
+      if (current >= total) { return false;}
 
       try {
         checkRead();
-        currentValue = recordReader.read();
+
+        corruptRecordCounter.incRecordsSeen();
+        boolean corrupt = false;
+
+        try {
+          currentValue = recordReader.read();
+        } catch (CorruptRecordException e) {
+          // this might throw, but it's fatal if it does.
+          corruptRecordCounter.incErrorsSeen(e);
+          corrupt = true;
+        }
+
         current ++;
+
+        if (corrupt) {
+          if (DEBUG) LOG.debug("skipping a corrupt record");
+          continue;
+        }
```

Comment:
I would agree that the control flow could be simplified rather than the try{} cacth {} setting a boolean that triggers a continue later.